### PR TITLE
initial implementation of front end

### PR DESCRIPTION
**Pull request by wademauger**

### CODE REVIEW

File: app/js/components/tags.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    render() {
+      let tags = this.props.tags;
+      if (typeof tags !== 'undefined'){
+        tags.sort();
+      }else{
+        tags = [];
+      }   
+      return(
+              <div>
+              {tags.map(function makeTags(thisTag) {
```
Comment (djrenren) [inline app/js/components/tags.js]: Use an arrow function here and there's no need to prefix the key

``` javascript
tags.map((thisTag) => <Tag key={thisTag} tag={thisTag} />);
```

File: app/js/components/tags.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+import React, { Component } from 'react';
+import Tag from './tag';
+
+export default class Tags extends Component {
```
Comment (djrenren) [inline app/js/components/tags.js]: Rename to TagList

**Comment by djrenren** [comment]
In general, I think we should look at using propTypes: https://facebook.github.io/react/docs/reusable-components.html#es6-classes so that we can get really nice runtime errors if we pass the wrong type of params around.

File: app/js/components/tag.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+import React, { Component } from 'react';
+
+export default class Tag extends Component {
+    render() {
+      return(
+            <button className='qdb-tag-button btn btn-primary btn-sm' id='tag-{this.props.tag}'>{this.props.tag}</button>
```
Comment (djrenren) [inline app/js/components/tag.js]: I don't think that syntax for `id=` would work...

Make sure the id is working. I think you'd need `id={`tag-${this.props.tag}`}`

**Comment by wademauger** [comment]
Why does this fail?

File: app/js/components/tags-list.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+export default class TagsList extends Component {
+    render() {
+      let tags = this.props.tags;
+      if (typeof tags !== 'undefined') {
+        tags.sort();
+      } else {
+        tags = [];
+      }
+      return(
+              <div>
```
Comment (kristenmills) [inline app/js/components/tags-list.js]: Indentation fix

**Comment by bbesmanoff** [comment]
Squash readme changes.

File: app/js/components/new-quote-form.js [hunk 5]
```diff
@@ -0,0 +1,21 @@
+'use strict';
+
+import React, { Component } from 'react';
+
+export default class QuoteForm extends Component {
```
Comment (bbesmanoff) [inline app/js/components/new-quote-form.js]: `CreateQuoteForm`

File: app/js/components/tags-list.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+'use strict';
+
+import React, { Component } from 'react';
+import Tag from './tag';
+
+export default class TagsList extends Component {
```
Comment (bbesmanoff) [inline app/js/components/tags-list.js]: `TagList`

File: app/js/components/tags-list.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+    render() {
+      let tags = this.props.tags;
```
Comment (bbesmanoff) [inline app/js/components/tags-list.js]:
``` javascript
let tags = this.props.tags || [];
tags.sort();
```

File: app/js/components/qdb.js [hunk 8]
```diff
@@ -1,8 +1,15 @@
...
 
+import Navbar from './navbar';
+
 export default class QDB extends React.Component {
   render() {
-    return <h1>QDB</h1>;
+    return(
+        <div>
+            <Navbar />
+               {this.props.children}
```
Comment (bbesmanoff) [inline app/js/components/qdb.js]: Indentation

File: app/js/components/quotes-page.js [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+'use strict';
+
+import React from 'react';
+
+import TagsList from './tags-list';
+import Quote from './quote';
+
+export default class QuotesPage extends React.Component {
+    render() {
+      let quoteslist = this.props.quotes;
```
Comment (bbesmanoff) [inline app/js/components/quotes-page.js]:
``` javascript
let quotesList = this.props.quotes || [];
```

File: app/js/components/quotes-page.js [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+      if (typeof quoteslist === 'undefined') {
+        quoteslist = [];
+      }
+      return(
+        <div>
+            <div className='container' id='qdb-content'>
+                <div className='page-header'>
+                  <h1>Quotes</h1>
+                </div>
+                    {quoteslist.map(function putQuotes(thisQuote) {
```
Comment (bbesmanoff) [inline app/js/components/quotes-page.js]:
```
.map((thisQuote) => {
  ...
}
```

File: app/js/components/tags-list.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+      if (typeof tags !== 'undefined') {
+        tags.sort();
+      } else {
+        tags = [];
+      }
+      return(
+              <div>
+              tags.map((thisTag) => <Tag key={'qdb-tag-' + thisTag}  tag={thisTag} />);
```
Comment (bbesmanoff) [inline app/js/components/tags-list.js]:
``` jsx
{tags.map((thisTag) => <Tag key={`qdb-tag${thisTag}`}  tag={thisTag} />)}
```

File: app/js/components/tag.js [hunk 12]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+import React, { Component } from 'react';
+
+export default class Tag extends Component {
+    render() {
+      return(
+            <button className='qdb-tag-button btn btn-primary btn-sm' id={'qdb-tag-{this.props.tag}'}>{this.props.tag}</button>
```
Comment (bbesmanoff) [inline app/js/components/tag.js]:
``` javascript
id={`qdb-tag-${this.props.tag}`}
```

File: app/js/components/add-quote-page.js [hunk 13]
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+import React from 'react';
+
+import QuoteForm from './new-quote-form';
+
+export default class QuoteFormPage extends React.Component {
+  render() {
+    return(
+        <div>
```
Comment (bbesmanoff) [inline app/js/components/add-quote-page.js]: Mixed 2/4 space indent

File: app/js/components/tag.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+import React, { Component } from 'react';
+
+export default class Tag extends Component {
+    render() {
+      return(
+            <button className='qdb-tag-button btn btn-primary btn-sm' id={'qdb-tag-${this.props.tag}'}>{this.props.tag}</button>
```
Comment (bbesmanoff) [inline app/js/components/tag.js]:
``` javascript
id={`qdb-tag-${this.props.tag}`}
```

File: app/js/components/tags-page.js [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+*to display them.
+* */
+
+'use strict';
+
+import React from 'react';
+
+import TagsList from './tags-list';
+
+export default class QDB extends React.Component {
```
Comment (bbesmanoff) [inline app/js/components/tags-page.js]: `class TagsPage`

**Comment by bbesmanoff** [comment]
:+1:

**Pull request merged**